tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -7108,7 +7108,7 @@ void CGMSHLSLRuntime::EmitHLSLOutParamConversionInit(
         !isObject) {
       QualType ArgTy = Arg->getType();
       Value *outVal = nullptr;
-      bool isAggregateTy = ParamTy->isAggregateType() && !IsHLSLVecMatType(ParamTy);
+      bool isAggregateTy = hlsl::IsHLSLAggregateType(ParamTy);
```

Comment:
@tex3d , sorry this touches a few lines of the outparamconversion code. Let me know if I should wait before checking this in.